src/pages/api/csvZoneUpload.ts

```diff
@@ -178,8 +178,12 @@ export default async (req: NextApiRequestWithSession, res: NextApiResponse): Pro
                 setFareZoneAttributeAndRedirect(req, res, errors);
                 return;
             }
-            const fareZoneName = userFareZones[0].FareZoneName;
-            await putDataInS3(userFareZones, `${uuid}.json`, true);
+            const fareZoneName = userFareZones[0].FareZoneName.replace('/', '');
+            const nocCode = getAndValidateNoc(req, res);
+            if (!nocCode) {
+                throw new Error('Could not retrieve nocCode from ID_TOKEN_COOKIE');
+            }
+            await putDataInS3(userFareZones, `fare-zone/${nocCode}/${fareZoneName}-${uuid}.json`, true);
```

Comment:
This is retrieved later on to generate the matching data so we'll probably need to update the path for that as well